src/detect-parse.c

```diff
@@ -1452,12 +1467,7 @@ static int SigParse(DetectEngineCtx *de_ctx, Signature *s,
         SCReturnInt(-1);
     }
 
-    s->sig_str = SCStrdup(sigstr);
-    if (unlikely(s->sig_str == NULL)) {
-        SCReturnInt(-1);
-    }
-
-    int ret = SigParseBasics(de_ctx, s, sigstr, parser, addrs_direction);
+    int ret = SigParseBasics(de_ctx, s, sigstr, parser, addrs_direction, requires);
```

Comment:
Is there no overhead for the parsing of tokens done twice per rule and creating duplicates of the sigstr?

Reply:
There is, as it does run through the rules 2x. But its very fast.  On my desktop the loading of 68693 rules took 36 seconds. The total time spent in this first pass was only 0.78 seconds of that.  The actual "parsing" of the rules is very fast, its the `Setup()` functions that take most of the time which is skipped in the first pass, except for `requires` and `sid`. This was with ASAN enabled.

The overhead is more present on a slow Raspberry PI where the prepares takes 1.4 seconds, and the total parse is 9 seconds (no ASAN this time).